### PR TITLE
Add two fields to dart context.

### PR DESCRIPTION
**Pull request by davidmorgan**
strong_analysis, for output of the analyzer, and checks, for the result of post processing on that output.

### CODE REVIEW

File: dart/build_rules/common/context.bzl [hunk 1]
```diff
@@ -43,6 +44,8 @@ def make_dart_context(
     package = None,
     lib_root = None,
     enable_summaries = False,
+    enable_analysis = False,
+    checks = None,
```
Comment (jakemac53) [inline dart/build_rules/common/context.bzl]: Would you mind adding some comments below for all the args, the list is getting pretty long now and these ones are a little less well known that some of the others.

File: dart/build_rules/common/context.bzl [hunk 2]
```diff
@@ -120,6 +131,8 @@ def _merge_dart_context(dart_ctx1, dart_ctx2):
       package = dart_ctx1.package,
       lib_root = dart_ctx1.lib_root,
       strong_summary = dart_ctx1.strong_summary,
+      strong_analysis = dart_ctx1.strong_analysis,
+      checks = dart_ctx1.checks,
```
Comment (jakemac53) [inline dart/build_rules/common/context.bzl]: It feels like we should have a TODO here or something, this is consistent with handling of `strong_summary` but it seems incorrect. I think in reality in the context that these merged contexts are used these fields aren't accessed, so we could assign these to `None` possibly, or even omit the fields entirely. Not something you need to do in this PR though.

Comment (davidmorgan) [inline dart/build_rules/common/context.bzl]: Added a TODO.

File: dart/build_rules/common/context.bzl (hunk 1 above)
Comment (davidmorgan) [inline dart/build_rules/common/context.bzl]: Done.

**Comment by davidmorgan** [comment]
Thanks, PTAL.

File: dart/build_rules/common/context.bzl [hunk 3]
```diff
@@ -43,11 +44,29 @@ def make_dart_context(
...
     license_files = []):
-  """Creates a dart context for a target."""
+  """Creates a dart context for a target.
+
+  Args:
+    ctx: The target context.
+    package: The dart package name.
+    lib_root: The lib folder for the package.
+    enable_summaries: Whether to generate analyzer summaries.
+    enable_summaries: Whether to generate analyzer output.
```
Comment (jakemac53) [inline dart/build_rules/common/context.bzl]: enable_analysis :)

**Comment by davidmorgan** [comment]
Done.

**Pull request merged**